### PR TITLE
[PM-1953] Vendor openSSL dependency (static linking)

### LINKED ISSUE #75: Unable to find libssl.so.3
### Steps To Reproduce

1. Download latest linux release on Ubuntu 20.04
1. Execute bws

### Expected Result

No error

### Actual Result

bws: error while loading shared libraries: libssl.so.3: cannot open shared object file: No such file or directory

### Screenshots or Videos

_No response_

### Additional Context

openssl and libssl-dev system packages are installed

### Operating System

Linux

### Operating System Version

Ubuntu 20.04

### Shell

Bash

### Build Version

N/A (latest release build)

### Issue Tracking Info

- [X] I understand that work is tracked outside of Github. A PR will be linked to this issue should one be opened to address it, but Bitwarden doesn't use fields like "assigned", "milestone", or "project" to track progress.

### PR DESCRIPTION
**Pull request by Hinton**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

- Cross-compile aarch64-unknown-linux-gnu using cross.
- Statically link openSSL by marking it as a vendored dependency.

Resolves #75, #31.

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)

### CODE REVIEW

File: .github/workflows/build-cli.yml [hunk 1]
```diff
@@ -67,21 +70,27 @@ jobs:
...
-          sudo apt-get install gcc-aarch64-linux-gnu
-
       - name: Cache cargo registry
         uses: Swatinem/rust-cache@dd05243424bd5c0e585e4b55eb2d7615cdd32f1f # v2.5.1
         with:
           key: ${{ matrix.settings.target }}-cargo-${{ matrix.settings.os }}
 
+      - name: Install Cross (aarch64-unknown-linux-gnu)
+        if: ${{ matrix.settings.target == 'aarch64-unknown-linux-gnu' }}
+        run: cargo install cross
```
Comment (MGibson1) [inline .github/workflows/build-cli.yml:80]: Is this cached as part of cargo caching?

File: crates/bws/Cargo.toml
Comment (MGibson1) [inline crates/bws/Cargo.toml:1]: can't say I love this formatter. Is it documented so we're consistent at least?

File: crates/bws/Cargo.toml [hunk 2]
```diff
@@ -15,23 +15,31 @@ keywords = ["bitwarden", "secrets-manager", "cli"]
...
 toml = "0.7.4"
-comfy-table = "^7.0.1"
-chrono = { version = "0.4.26", features = ["clock", "std"], default-features = false }
 uuid = { version = "^1.3.3", features = ["serde"] }
 
 [dev-dependencies]
 tempfile = "3.5.0"
+
+[target.'cfg(target_os = "linux")'.dependencies]
+openssl = { version = "0.10", features = ["vendored"] }
```
Comment (MGibson1) [inline crates/bws/Cargo.toml:45]: Statically linking this probably activates more of the [OpenSSL license ](https://www.openssl.org/source/license-openssl-ssleay.txt) requirements -- namely, that you have to be able to get at the copyright from the binary and any marketing we do needs to credit OpenSSL

File: crates/bws/Cargo.toml
Comment (Hinton) [reply crates/bws/Cargo.toml:1]: It's the vscode TOML extension.

File: .github/workflows/build-cli.yml (hunk 1 above)
Comment (Hinton) [reply .github/workflows/build-cli.yml:80]: Yes

File: crates/bws/Cargo.toml (hunk 2 above)
Comment (Hinton) [reply crates/bws/Cargo.toml:45]: I think so, we could look into rusttls which gets rid of the openssl dependency.

Comment (Hinton) [reply crates/bws/Cargo.toml:45]: OpenSSL 3 is also licensed under Apache so we should be fine.

**Review by vgrassia — DISMISSED** [review]

File: crates/bws/Cargo.toml (hunk 2 above)
Comment (MGibson1) [reply crates/bws/Cargo.toml:46]: yes, I had 1.1.1 in my head

**Review by MGibson1 — APPROVED** [review]

**Pull request merged**